Unity-Technologies/URDF-Importer
# Better execption handling in runtime mode when 'robot' tag is not added

## Proposed change(s)

This is not actual bug fix, but is to provide more information so that a user knows 'robot' tag is missing and he/she has to add a 'robot' tag in Project Settings in the Unity Editor. This doesn't change the behavior of the URDF importer in Editor as tags can be added in Editor. However since there is no way to add or query tags in a stand alone build outside Unity Editor, without the proposed changes, the Log will be flooded with NullReferenceExceptions every frame without proper information about the root cause.

- Added exception handling to print more informative information when 'robot' tag is not found (stand alone build). 
- Also minor code factoring to make the shared 'robot' tag string a constant and reused.

### Useful links (GitHub issues, JIRA tickets, forum threads, etc.)
https://jira.unity3d.com/browse/AIRO-973

### Types of change(s)


## Testing and Verification

 1. Create an Emply Unity Scene.
 1. Check the Project Settings > Tags to make sure there is 'robot' tag. (if there is one, remove)
 1. Add an empty GameObject and add a RuntimeURDFImporter (or if renamed, RuntimeURDFImporterExample)
 1. Create a standalone build (mac or windows) with Development Option On
 1. Run the build, 
 1. In the GUI, add the full path to a URDF file, and Click on Load
 1. You should notice error logs stating that the robot tag needs to be added in Unity Editor

(Optional)
 1. repeat the above, but this time add a 'robot' tag in the project settings, before creating the build. The URDF file should be loaded without an issue.


### Test Configuration:
- Unity Version: 2020.3.13f1
- Unity machine OS + version: Windows 10 or MacOS

## Checklist
- [x] Ensured this PR is up-to-date with the `dev` branch
- [x] Created this PR to target the `dev` branch
- [x] Followed the style guidelines as described in the [Contribution Guidelines](https://github.com/Unity-Technologies/URDF-Importer/blob/main/CONTRIBUTING.md)
- [ ] Increased the [test coverage criteria](https://github.com/Unity-Technologies/URDF-Importer/blob/dev/.yamato/yamato-config.yml#L18) by 3%
- [x] Updated the [Changelog](https://github.com/Unity-Technologies/URDF-Importer/blob/dev/com.unity.robotics.urdf-importer/CHANGELOG.md) and described changes in the [Unreleased section](https://github.com/Unity-Technologies/URDF-Importer/blob/dev/com.unity.robotics.urdf-importer/CHANGELOG.md#unreleased)
- [ ] Updated the documentation as appropriate
